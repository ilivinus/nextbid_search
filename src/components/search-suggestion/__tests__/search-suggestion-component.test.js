import React from 'react';
import renderer from 'react-test-renderer';
import SearchSuggestionComponent from '../search-suggestion-component';

describe("#SearchSuggestionComponent",() => {

    it("Should take the snapshot of the component",()=>{
        
        let data = ["Marnix","Livinus","Chidi","Peter"]
        ,getSelectionLength = (len) =>{}
        ,suggestionClick = jest.fn()
        ,searchKey= "searching";

        let component = renderer.create(
            <SearchSuggestionComponent
                searchKey={searchKey}
                getSelectionLength={getSelectionLength}
                suggestionClick={suggestionClick}
                suggested_items={[]} />)
        let tree =  component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("Should call the appropriate prop",()=>{

        let data = ["Marnix","Livinus","Chidi","Peter"]
        ,getSelectionLength = (len) =>{}
        ,suggestionClick = jest.fn()
        ,searchKey= "searching";

        let component = renderer.create(
            <SearchSuggestionComponent
                searchKey={searchKey}
                getSelectionLength={getSelectionLength}
                suggestionClick={suggestionClick}
                suggested_items={data} />)

        let tree = component.toJSON();
        let props = tree.children;
          
        //manually trigger callback
        props[0].props.onClick();
        expect(suggestionClick).toHaveBeenCalledTimes(1);       

        expect(tree.children.length).toBe(4);
    })
});