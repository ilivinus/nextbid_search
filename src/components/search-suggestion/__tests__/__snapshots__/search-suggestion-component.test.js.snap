// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SearchSuggestionComponent Should take the snapshot of the component 1`] = `
<div
  className="suggestion-items"
/>
`;
